File: Swift/프로그래머스/1/42889. 실패율/실패율.swift
import Foundation

func solution(_ N:Int, _ stages:[Int]) -> [Int] {
    
    var map = Array(repeating: 0, count: N+2)
    for stage in stages {
        map[stage] += 1
    }
    
    let total_person = stages.count
    var cnt = 0
    var ans: [(Int, Double)] = []
    
    for (i, m) in map.enumerated() {
        if i != 0 && i <= N {
            var cur_all_person = total_person - cnt
            var not_pass_person = m
            
            cnt += m
            
            if cur_all_person == 0 {
                ans.append((i, 0.0))
            } else {
                ans.append((i, Double(not_pass_person) / Double(cur_all_person)))
            }
        } 
    }
    
    var result: [Int] = []
    var sorted_ans = ans.sorted { $0.1 > $1.1 }

    for a in sorted_ans {
        result.append(a.0)
    }
    
    return result
}

File: Python3/프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    persons = len(stages)
    distance = [0 for _ in range(N+2)]
    
    for idx, stage in enumerate(stages):
        distance[stage] += 1
    
    cnt = 0
    ans = []
    for idx in range(1,N+1):
        total = persons - cnt
        non_pass = distance[idx]
        cnt += non_pass
        
        if total != 0:
            ans.append((idx, non_pass/total))
        else:
            ans.append((idx, 0))
        
    ans.sort(key= lambda x: (-x[1], x[0]))
    
    result = []
    for a in ans:
        result.append(a[0])
        
    return result